gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -540,16 +542,25 @@ void connectHelixManagerWithRetry() {
    * the job with EXAMPLE_INSTANCE_TAG will remain in the ZK until an instance with EXAMPLE_INSTANCE_TAG was found.
    */
   private void addInstanceTags() {
-    List<String> tags = ConfigUtils.getStringList(this.clusterConfig, GobblinClusterConfigurationKeys.HELIX_INSTANCE_TAGS_KEY);
     HelixManager receiverManager = getReceiverManager();
     if (receiverManager.isConnected()) {
-      if (!tags.isEmpty()) {
-        logger.info("Adding tags binding " + tags);
-        tags.forEach(tag -> receiverManager.getClusterManagmentTool()
-            .addInstanceTag(this.clusterName, this.helixInstanceName, tag));
-        logger.info("Actual tags binding " + receiverManager.getClusterManagmentTool()
-            .getInstanceConfig(this.clusterName, this.helixInstanceName).getTags());
+      // The helix instance associated with this container should be consistent on helix tag
+      List<String> existedTags = receiverManager.getClusterManagmentTool()
+          .getInstanceConfig(this.clusterName, this.helixInstanceName).getTags();
+      Set<String> desiredTags = new HashSet<>(
+          ConfigUtils.getStringList(this.clusterConfig, GobblinClusterConfigurationKeys.HELIX_INSTANCE_TAGS_KEY));
+      if (!desiredTags.isEmpty()) {
+        // If a helix instance already have tag assigned during last run, it won't be auto removed. Need to remove unwanted tags
```

Comment:
Modify comment to: "Remove tag assignments for the current Helix instance from a previous run."

Reply:
done